routes/api/contacts.js

```diff
@@ -1,25 +1,76 @@
-const express = require('express')
+const Joi = require("joi");
+const express = require("express");
+const router = express.Router();
+const contacts = require("../../models/contacts");
 
-const router = express.Router()
+const schema = Joi.object({
+  name: Joi.string().min(3).max(20),
+  email: Joi.string().email({ minDomainSegments: 2 }),
+  phone: Joi.string().min(9),
+});
 
-router.get('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/", async (req, res, next) => {
+  const data = await contacts.listContacts();
+  res.json(data);
+});
 
-router.get('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.get("/:contactId", async (req, res, next) => {
+  const contactId = req.params.contactId;
+  const dataID = await contacts.getContactById(contactId);
+  if (!dataID) {
+    res.status(404).json({ message: "Not found." });
+  }
+  res.status(200).json(dataID);
+});
 
-router.post('/', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+router.post("/", async (req, res, next) => {
+  const { name, email, phone } = req.body;
+  const { error } = schema.validate(req.body);
+  if (error) {
+    return res.status(400).send(error.details[0].message);
+  }
 
-router.delete('/:contactId', async (req, res, next) => {
-  res.json({ message: 'template message' })
-})
+  if (!name || !email || !phone) {
```

Comment:
Użyj joi do tego, jak dodasz tam require to nie powinnaś potrzebować tego miejsca